### PR TITLE
App: refactor license handling to reduce code duplication

### PR DESCRIPTION
**Pull request by wwmayer**
Thank you for creating a pull request to contribute to FreeCAD! Place an "X" in between the brackets below to "check off" to confirm that you have satisfied the requirement, or ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10) if there is something you don't understand.

- [x]  Your Pull Request meets the requirements outlined in section 5 of [CONTRIBUTING.md](https://github.com/FreeCAD/FreeCAD/blob/master/CONTRIBUTING.md) for a Valid PR

Please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [1.0 Changelog Forum Thread](https://forum.freecad.org/viewtopic.php?f=10&t=69438).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8458">pipeline status</a> for feature branch [PR_8458](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8458). Pipeline [775262948](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775262948) was triggered at [7ec49b7](https://github.com/FreeCAD/FreeCAD/pull/8458/commits/7ec49b7e51d8e9488e1e27f35206e9887fce6b78). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8458">pipeline status</a> for feature branch [PR_8458](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_8458). Pipeline [775272816](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/775272816) was triggered at [15b15b1](https://github.com/FreeCAD/FreeCAD/pull/8458/commits/15b15b190a522d25cf603610579cf9b644f7aeb0). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Gui/DlgProjectInformationImp.cpp [hunk 1]
```diff
@@ -29,11 +29,28 @@
...
 
 #include <App/Document.h>
+#include <App/License.h>
 
 #include "DlgProjectInformationImp.h"
 #include "ui_DlgProjectInformation.h"
 
 
+#if 0 // needed for Qt's lupdate utility
+    qApp->translate("Gui::Dialog::DlgSettingsDocument", "All rights reserved");
```
Comment (0penBrain) [inline src/Gui/DlgProjectInformationImp.cpp:39]: Why not do this in 'License::init()' using 'QT_TRANSLATE_NOOP'?

**Comment by donovaly** [comment]
This PR does not compile here on Windows.

I have to add an include  -> will make a PR

But with this, I still get this warning:
`4>D:\FreeCADGit\src\App/License.h(75,40): warning C4251: 'App::License::licenseItems': class 'std::map<App::License::Type,App::License::LicenseItem,std::less<App::License::Type>,std::allocator<std::pair<const App::License::Type,App::License::LicenseItem>>>' needs to have dll-interface to be used by clients of class 'App::License'
4>D:\FreeCADGit\src\App/License.h(75,17): message : see declaration of 'std::map<App::License::Type,App::License::LicenseItem,std::less<App::License::Type>,std::allocator<std::pair<const App::License::Type,App::License::LicenseItem>>>'`

**Comment by wwmayer** [comment]
IIRC to fix this warning the trick was to remove the AppExport macro from the class and add it to every public function of the class.

File: src/Gui/DlgProjectInformationImp.cpp (hunk 1 above)
Comment (wwmayer) [reply src/Gui/DlgProjectInformationImp.cpp:39]: Because i18n is not yet supported in App modules. Once it's done the code can move to License.cpp

Comment (0penBrain) [reply src/Gui/DlgProjectInformationImp.cpp:39]: Thanks for insight.

**Pull request merged**